src/main/java/com/tjtanjin/steve/parser/Parser.java

```diff
@@ -0,0 +1,229 @@
+    }
+
+    /**
+     * Parses input from user to determine action to take.
+     *
+     * @param input input provided by user
+     * @return string response after operation is done
+     */
+    public String parseInput(String input) {
+
+        String[] parsedInput = input.split(" ", 2);
+
+        String command = parsedInput[0].toUpperCase();
+        if (!isValidCommand(command)) {
+            return "Error: Invalid instruction, type 'help' for more options";
+        }
+
+        try {
+            switch (Cmd.valueOf(command)) {
+            case BYE:
+                return BYE_COMMAND.execute();
+            case DEADLINE:
+                String deadlineTaskName = parseTaskName(input);
+                LocalDate[] deadlineTaskDates = parseTaskDates(input);
+                return DEADLINE_COMMAND.execute(command, deadlineTaskName, deadlineTaskDates);
+            case DELETE:
+                int deleteIndex = parseIndex("delete", input);
+                return DELETE_COMMAND.execute(deleteIndex);
+            case DONE:
+                int doneIndex = parseIndex("done", input);
+                return DONE_COMMAND.execute(doneIndex);
+            case EVENT:
+                String eventTaskName = parseTaskName(input);
+                LocalDate[] eventTaskDates = parseTaskDates(input);
+                return EVENT_COMMAND.execute(command, eventTaskName, eventTaskDates);
+            case FIND:
+                String searchTerm = parseTaskName(input);
+                return FIND_COMMAND.execute(searchTerm);
+            case HELP:
+                return HELP_COMMAND.execute();
+            case LIST:
+                return LIST_COMMAND.execute();
+            case TODO:
+                String todoTaskName = parseTaskName(input);
+                return TODO_COMMAND.execute(command, todoTaskName);
+            default:
+                return "Error: Invalid instruction, type 'help' to see the options.";
+            }
+        } catch (SteveInvalidFormatException | SteveInvalidParamsException e) {
+            return e.getMessage();
+        }
+    }
+
+    /**
+     * Checks if a command provided by user is valid.
+     *
+     * @param command command provided by user
+     * @return true if command is valid, false otherwise
+     */
+    private boolean isValidCommand(String command) {
+        Cmd[] cmds = Cmd.values();
+        for (Cmd cmd: cmds) {
+            if (cmd.name().equals(command)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Parses index from user for done and delete command.
+     *
+     * @param cmd cmd from user (done or delete)
+     * @param input input provided by user
+     * @return index of task
+     */
+    private int parseIndex(String cmd, String input)
+            throws SteveInvalidFormatException, SteveInvalidParamsException {
+        String[] parsedString = input.split("\\s+");
```

Comment:
parsedStrings?

Reply:
Similar to the previous comment, I had opted to use a singular form in `parsedString` to indicate that what was parsed was the `input` argument (although perhaps `parsedInput` would sound more related). Would this sound appropriate or intuitive?